### ISSUE TITLE
[ECS] [parameter get all keys by path]: secrets from parameter store just provide a path

### ISSUE THREAD

**Issue by nodesocket** [aws/containers-roadmap#535]
In ECS and the service definition I use secrets stored in parameter store like:

```
"secrets":[
      {
         "valueFrom":"arn:aws:ssm:us-west-1:XXXXXXXXXXX:parameter/WEB_APP/DB_PASSWORD",
         "name":"DB_PASSWORD"
      },
      {
        "valueFrom": "arn:aws:ssm:us-west-1:XXXXXXXXXXX:parameter/WEB_APP/DB_USERNAME",
        "name": "DB_USERNAME"
      },
      {
        "valueFrom": "arn:aws:ssm:us-west-1:XXXXXXXXXXX:parameter/WEB_APP/DB_HOST",
        "name": "DB_HOST"
      },
     ...
]
````

We have lots of secrets, so defining all them in the service definition JSON becomes tedious and very error prone. Instead, I should just be able to define a **path** and ECS should pull down all keys in that path and use them.

Example of how that could look:

```
"secrets":[
      {
         "valueFrom":"arn:aws:ssm:us-west-1:XXXXXXXXXXX:parameter/WEB_APP/*",
         "name":"$KEY_NAME"
      }
]
````

**Comment by vegardx** [comment]
This would be very helpful for making more generic terraform modules.

**Comment by dayer4b** [comment]
Yes!  Please!  We are avoiding using the `secrets[].valueFrom` thing basically for lack of this feature.  Engineers have pushed back that the list of parameters here effectively represents a second source of truth.  Instead we rely on an entrypoint script that does the `get-parameters-by-path` thing.

However, the entrypoint script brings with it a limitation that many of us would like to ditch.

**Comment by nodesocket** [comment]
Any update on this? Allowing wildcard pulling of a path from parameter store would be a huge win in our Terraform.

**Comment by dayer4b** [comment]
I just noticed that this seems very similar to issue #246.  In that issue, the reporter mentions a tool called [ssm-parent](https://github.com/springload/ssm-parent).  I haven't tried it, but that may help some people.

**Comment by anagarjunr** [comment]
We are facing the same issue.
would love to see this for our fargate containers and greatly reduces our ops complexity

**Comment by phillipuniverse** [comment]
There is a very similar tool to `ssm-parent` called [chamber](https://github.com/segmentio/chamber) which we are using in the meantime.

**Comment by alexjeen** [comment]
This would be so helpful, especially since now every env change requires a taskdef update and this a new deployment to make it work.
